challengeutils/synapseservices/challenge.py

```diff
@@ -0,0 +1,102 @@
+"""Challenge Services"""
+from .base_service import Service
+
+
+class Challenge(Service):
+    """Synapse Challenge object
+    https://docs.synapse.org/rest/org/sagebionetworks/repo/model/Challenge.html
+
+    Args:
+        id: The ID of this Challenge object
+        etag: Synapse employs an Optimistic Concurrency Control (OCC)
+              scheme to handle concurrent updates.
+        projectId: The ID of the Project the challenge is used with.
+        participantTeamId: The ID of the Team which users join to participate
+                           in the Challenge
+    """
+    def __init__(self, id: str = None, projectId: str = None, etag:
+                 str = None, participantTeamId: str = None):
+
+        self.openapi_types = {
+            'id': str,
+            'projectId': str,
+            'etag': str,
+            'participantTeamId': str
+        }
+
+        self.attribute_map = {
```

Comment:
I don't think I understand the purpose of this...

Reply:
You can see the purpose of this in the `base_service.py`.  Essentially a lot of API's use camelCase, but a lot of python now uses snakecase.
```
        attribute_map: The key is attribute name and the
                       value is json key in definition.
```

I originally had:
```
        self.attribute_map = {
            'id': 'id',
            'projectid': 'projectId',
            'etag': 'etag',
            'participant_teamid': 'participantTeamId'
        }
```

Which then wouldn't throw the pep8 warning:
```
Attribute name "projectId" doesn't conform to snake_case naming style
```

That being said, I still wanted people to be able to create the class with the response object, so I stuck with camelCase.

Reply:
Oh I see now. Got it :thumbsup: